From: Geoff Gustafson <grgustaf@redhat.com>
Subject: [RHEL5 PATCH] sata ata_piix map values (bz 204684)
Date: Fri, 01 Dec 2006 21:49:07 -0500
Bugzilla: 204684
Message-Id: <4570E9A3.1010209@redhat.com>
Changelog: sata ata_piix map values


This patch corrects the ICH8 map values for sata ata_piix. This fixes a 
problem where SATA devices are not detected on some of the SATA ports on 
an ICH8 motherboard when the BIOS is set to IDE mode.

I tested with the procedure described in the bug: installed with IDE 
setting in BIOS, found a SATA connector where the DVD drive was not 
shown in /proc/scsi/scsi under the old kernel. Booted with the patched 
kernel and the drive was detected. I did this based off the 1.2767.el5 
kernel SRPM.

Patch is upstream here:
http://www.kernel.org/git/?p=linux/kernel/git/torvalds/linux-2.6.git;a=blobdiff;h=5250187ffce2a610f456e3814d019bc3a4b656b6;hp=5719704eb0ee5a6d5699399cf2edb3b12f174f18;hb=cb7fabcf9d4340b61f012cc7e8e3d7ff5c4f441c;f=drivers/ata/ata_piix.c

Please ACK.

- Geoff



Index: linux-2.6.18.x86_64/drivers/scsi/ata_piix.c
===================================================================
--- linux-2.6.18.x86_64.orig/drivers/scsi/ata_piix.c	2006-09-19 23:42:06.000000000 -0400
+++ linux-2.6.18.x86_64/drivers/scsi/ata_piix.c	2006-12-01 21:41:48.000000000 -0500
@@ -361,9 +361,9 @@ static const struct piix_map_db ich8_map
 	.present_shift = 8,
 	.map = {
 		/* PM   PS   SM   SS       MAP */
-		{  P0,  NA,  P1,  NA }, /* 00b (hardwired) */
+		{  P0,  P2,  P1,  P3 }, /* 00b (hardwired when in AHCI) */
 		{  RV,  RV,  RV,  RV },
-		{  RV,  RV,  RV,  RV }, /* 10b (never) */
+		{  IDE,  IDE,  NA,  NA }, /* 10b (IDE mode) */
 		{  RV,  RV,  RV,  RV },
 	},
 };

